internal/server/kong/ws/config/compat/compatibility_table.go

```diff
@@ -834,6 +836,42 @@ var (
 				},
 			},
 		},
+		{
+			Metadata: config.ChangeMetadata{
```

Comment:
Why are these needed since we already have `P122` and `P123`? 

I think this can lead to some weird edge cases like this:
- `config.functions` is set to a non-null value
- `P135` is processed, removing the `functions` field
- `P122` doesn't get processed because `functions` is not set anymore, therefore `access` is not correctly populated

Right now there is no guarantee that updates are [processed in order](https://github.com/Kong/koko/pull/414)

Reply:
Oh I see why you're doing this.

Still, I think this can lead to some weird issue. What about short-circuiting the `ValueFromField` but not the field removal? This way we would keep the changes limited to how the `ValueFromField` is used, which is where the bug resides.

WDYT? (cc: @mikefero )

Reply:
After talking with @mikefero I think we are good on this front: the ordering issue **is** indeed an issue, but not for this specific case. 

Can you please make sure we have a ticket to track this though?

Reply:
I'm not really understanding, unfortunately. I'm pushing some updates to how this works based on review feedback which will impact expected behavior so do you mind revisiting on the next review pass?